### PR TITLE
RMP-11935 Option to create and name Azure Resource Group while provis…

### PR DESCRIPTION
**Pull request by topolyai5**

### CODE REVIEW

File: cloud-azure/src/main/java/com/sequenceiq/cloudbreak/cloud/azure/AzureSetup.java [hunk 1]
```diff
@@ -131,7 +133,7 @@ public ImageStatusResult checkImageStatus(AuthenticatedContext ac, CloudStack st
 
     @Override
     public void prerequisites(AuthenticatedContext ac, CloudStack stack, PersistenceNotifier persistenceNotifier) {
-        String storageGroup = azureUtils.getResourceGroupName(ac.getCloudContext());
+        String storageGroup = azureUtils.getResourceGroupName(ac.getCloudContext(), stack);
```
Comment (pdarvasi) [inline cloud-azure/src/main/java/com/sequenceiq/cloudbreak/cloud/azure/AzureSetup.java]: I would suggest using var resourceGroup here.

File: cloud-azure/src/main/java/com/sequenceiq/cloudbreak/cloud/azure/AzureSetup.java [hunk 2]
```diff
@@ -158,6 +160,20 @@ public void validateFileSystem(CloudCredential credential, SpiFileSystem spiFile
...
+    @Override
+    public void validateParameters(AuthenticatedContext ac, Map<String, String> parameters) throws Exception {
+        AzureClient client = ac.getParameter(AzureClient.class);
+        String resourceGroupName = parameters.get(AzureResourceConnector.RESOURCE_GROUP_NAME);
+        if (StringUtils.isEmpty(resourceGroupName)) {
+            return;
+        }
+        ResourceGroup resourceGroup = client.getResourceGroup(resourceGroupName);
+        if (resourceGroup != null) {
+            throw new BadRequestException("ResorceGroupName is exists: " + resourceGroup.name());
```
Comment (pdarvasi) [inline cloud-azure/src/main/java/com/sequenceiq/cloudbreak/cloud/azure/AzureSetup.java]: Suggestion: "Resourcegroup name already exists: "

Comment (doktoric) [reply cloud-azure/src/main/java/com/sequenceiq/cloudbreak/cloud/azure/AzureSetup.java]: or 'Resource group with {%name} already exists on provider side, please choose a different one.'

Comment (doktoric) [reply cloud-azure/src/main/java/com/sequenceiq/cloudbreak/cloud/azure/AzureSetup.java]: @pdarvasi what do you think ?

File: cloud-api/src/main/java/com/sequenceiq/cloudbreak/cloud/Setup.java [hunk 3]
```diff
@@ -54,6 +56,15 @@
      */
     void validateFileSystem(CloudCredential credential, SpiFileSystem spiFileSystem) throws Exception;
 
+    /**
+     * Implementation of this method shall validate the parameters if valid or not.
+     *
+     * @param authenticatedContext the context which already contains the authenticated client
+     * @param parameters map of paramters
```
Comment (doktoric) [inline cloud-api/src/main/java/com/sequenceiq/cloudbreak/cloud/Setup.java]: type paramters

File: core/src/main/java/com/sequenceiq/cloudbreak/controller/validation/ParametersValidator.java [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+    private EventBus eventBus;
+
+    @Inject
+    private ErrorHandlerAwareReactorEventFactory eventFactory;
+
+    public void validate(String platform, CloudCredential cloudCredential, Map<String, String> parameters, String userId, Long workspaceId) {
+        if (parameters == null || parameters.isEmpty()) {
+            return;
+        }
+        LOGGER.debug("Validate the file system: {}", parameters);
```
Comment (doktoric) [inline core/src/main/java/com/sequenceiq/cloudbreak/controller/validation/ParametersValidator.java]: is this a filesystem validation ?

File: cloud-api/src/main/java/com/sequenceiq/cloudbreak/cloud/Setup.java (hunk 3 above)
Comment (topolyai5) [reply cloud-api/src/main/java/com/sequenceiq/cloudbreak/cloud/Setup.java]: what is the type parameter?

**Pull request merged**